application_sandbox/extended_dynamic_state2/main.cpp

```diff
@@ -0,0 +1,412 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "application_sandbox/sample_application_framework/sample_application.h"
+#include "support/entry/entry.h"
+#include "vulkan_helpers/buffer_frame_data.h"
+#include "vulkan_helpers/helper_functions.h"
+#include "vulkan_helpers/vulkan_application.h"
+#include "vulkan_helpers/vulkan_model.h"
+
+#include <chrono>
+#include "mathfu/matrix.h"
+#include "mathfu/vector.h"
+
+using Mat44 = mathfu::Matrix<float, 4, 4>;
+using Vector4 = mathfu::Vector<float, 4>;
+
+namespace cube_model {
+#include "cube.obj.h"
+}
+
+// red cube
+const struct {
+  size_t num_vertices;
+  float positions[24];
+  float uv[16];
+  float normals[24];
+  size_t num_indices;
+  uint32_t indices[36];
+} cube_data = {8,
+               {0.5f,  0.5f,  0.5f,  0.5f,  -0.5f, 0.5f,  0.5f,  0.5f,
+                -0.5f, 0.5f,  -0.5f, -0.5f, -0.5f, 0.5f,  -0.5f, -0.5f,
+                -0.5f, -0.5f, -0.5f, 0.5f,  0.5f,  -0.5f, -0.5f, 0.5f},
+               {1.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f,
+                1.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f},
+               {0.5f,  0.5f,  0.5f,  0.5f,  -0.5f, 0.5f,  0.5f,  0.5f,
+                -0.5f, 0.5f,  -0.5f, -0.5f, -0.5f, 0.5f,  -0.5f, -0.5f,
+                -0.5f, -0.5f, -0.5f, 0.5f,  0.5f,  -0.5f, -0.5f, 0.5f},
+               36,
+               {0, 1, 3, 3, 2, 0, 1, 0, 6, 6, 7, 1, 4, 5, 7, 7, 6, 4,
+                2, 3, 5, 5, 4, 2, 0, 2, 4, 4, 6, 0, 3, 1, 7, 7, 5, 3}};
+
+// green cube
+const struct {
+  size_t num_vertices;
+  float positions[24];
+  float uv[16];
+  float normals[24];
+  size_t num_indices;
+  uint32_t indices[36];
+} cube_data2 = {8,
+                {1.0f,  0.5f,  0.5f,  1.0f,  -0.5f, 0.5f,  1.0f,  0.5f,
+                 -0.5f, 1.0f,  -0.5f, -0.5f, -1.0f, 0.5f,  -0.5f, -1.0f,
+                 -0.5f, -0.5f, -1.0f, 0.5f,  0.5f,  -1.0f, -0.5f, 0.5f},
+                {0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f,
+                 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f},
+                {0.5f,  0.5f,  0.5f,  0.5f,  -0.5f, 0.5f,  0.5f,  0.5f,
+                 -0.5f, 0.5f,  -0.5f, -0.5f, -0.5f, 0.5f,  -0.5f, -0.5f,
+                 -0.5f, -0.5f, -0.5f, 0.5f,  0.5f,  -0.5f, -0.5f, 0.5f},
+                36,
+                {0, 1, 3, 3, 2, 0, 1, 0, 6, 6, 7, 1, 4, 5, 7, 7, 6, 4,
+                 2, 3, 5, 5, 4, 2, 0, 2, 4, 4, 6, 0, 3, 1, 7, 7, 5, 3}};
+
+uint32_t cube_vertex_shader[] =
+#include "cube.vert.spv"
+    ;
+
+uint32_t cube_fragment_shader[] =
+#include "cube.frag.spv"
+    ;
+
+struct CubeFrameData {
+  containers::unique_ptr<vulkan::VkCommandBuffer> command_buffer_;
+  containers::unique_ptr<vulkan::VkFramebuffer> framebuffer_;
+  containers::unique_ptr<vulkan::DescriptorSet> cube_descriptor_set_;
+};
+
+VkPhysicalDeviceExtendedDynamicState2FeaturesEXT kFeat{
```

Comment:
I would add a note here that we are only enabling the basic extended_dynamic_state2 support, to use the depth bias element. The dynamic logic operation and patch control points are left off to allow the sample to be more broadly applicable.